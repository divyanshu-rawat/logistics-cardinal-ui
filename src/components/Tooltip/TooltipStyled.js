import styled from 'styled-components';

const TooltipStyled = styled('div')`
  position: relative;
  &:hover {
    &:after {
      opacity: 0.9;
    }
  }
  &:after {
    content: ${({ content }) => `'${content}'`};
    border-radius: ${({ theme }) => theme.borderRadius.mega};
    background: ${({ theme }) => theme.colors.grey400};
    border: 1px solid ${({ theme }) => theme.colors.grey500};
    color: ${({ theme }) => theme.colors.white};
    font-size: ${({ theme }) => theme.fontSize.bit};
    padding: 4px 8px;
    position: absolute;
    left: 50%;
    top: -6px;
    transform: translateX(-50%) translateY(-100%) scale(0);
    text-align: center;
    width: 100%;
    max-width: 280px;
    min-width: 120px;
    pointer-events: none;
    opacity: 0;
    display: block;
    transition: opacity 0.2s linear 0.2s, transform 0.2s ease-out 0.2s;
  }
  ${({ position }) => {
    switch (position) {
      case 'top':
        return `
          &:hover {
            &:after {
              transform: translateX(-50%) translateY(-100%) scale(1);
            }
          }
          &:after {
            margin-bottom: 10px;
            transform-origin: center bottom;
            left:50%;
          }
        `;
      case 'right':
        return `
          &:hover {
            &:after {
              transform: translateX(0%) translateY(-50%) scale(1);
            }
          }
          &:after {
            left:100%;
            top:50%;
            transform: translateX(-10%) translateY(-50%);
            margin-left: 10px;
            filter: blur(0);
            transform-origin: left center;
          }
        `;
      case 'bottom':
        return `
          &:hover {
            &:after {
              transform: translateX(-50%) translateY(0%) scale(1);
            }
          }
          &:after {
            top:100%;
            left:50%;
            margin-top: 10px;
            transform: translateX(-50%) translateY(-10%);
            transform-origin: center top;
          }
        `;
      case 'left':
        return `
          &:hover {
            &:after {
              transform: translateX(-100%) translateY(-50%) scale(1);
            }
          }
          &:after {
            left:0%;
            top:50%;
            transform: translateX(-90%) translateY(-50%);
            margin-left: -10px;
            filter: blur(0);
            transform-origin: right center;
          }
        `;
      default:
        return '';
    }
  }};
`;

export default TooltipStyled;
