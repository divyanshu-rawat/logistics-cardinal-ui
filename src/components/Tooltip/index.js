import React from 'react';
import PropTypes from 'prop-types';

import TooltipStyled from './TooltipStyled';

const Tooltip = ({ children, content, position }) => (
  <TooltipStyled content={content} position={position}>
    {children}
  </TooltipStyled>
);

Tooltip.propTypes = {
  /** Element to be wrapped into the tooltip */
  children: PropTypes.node.isRequired,
  /** The string content */
  content: PropTypes.string.isRequired,
  /** Tooltip Position */
  position: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
};

Tooltip.defaultProps = {
  position: 'top',
};

export default Tooltip;
