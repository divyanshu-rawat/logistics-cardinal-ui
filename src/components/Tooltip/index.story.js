import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { select } from '@storybook/addon-knobs';

import { COMPONENTS } from '../../../.storybook/sections';

import withTests from '../../utils/withTests';

import Tooltip from '.';
import Button from '../Button';

import Icons from '../Icons';

storiesOf(`${COMPONENTS}Tooltip`, module)
  .addDecorator(withTests('Tooltip/tests/index', 'TooltipStyled'))
  .add(
    'Simple',
    withInfo()(() => (
      <div>
        <Tooltip
          content="2017-32-32 12:12"
          position={select(
            'Position',
            ['top', 'right', 'bottom', 'left'],
            'top'
          )}
        >
          <Button>
            <Icons.Calendar />
          </Button>
        </Tooltip>
      </div>
    ))
  );
