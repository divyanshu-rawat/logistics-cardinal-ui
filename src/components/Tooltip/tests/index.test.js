import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Tooltip from '..';

const renderComponent = ({ content = 'Tooltip Text', position = 'top' }) =>
  shallow(
    <Tooltip content={content} position={position}>
      <span>Hover me</span>
    </Tooltip>,
  );

describe('<Tooltip />', () => {
  it('should render component with normal props', () => {
    const renderedComponent = renderComponent({});
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });

  it('should render component with position right', () => {
    const renderedComponent = renderComponent({ position: 'right' });
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });

  it('should render component with position bottom', () => {
    const renderedComponent = renderComponent({ position: 'bottom' });
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });

  it('should render component with position left', () => {
    const renderedComponent = renderComponent({ position: 'left' });
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
