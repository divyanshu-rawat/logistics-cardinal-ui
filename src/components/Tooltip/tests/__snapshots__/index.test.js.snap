// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render component with normal props 1`] = `
<TooltipStyled
  content="Tooltip Text"
  position="top"
>
  <span>
    Hover me
  </span>
</TooltipStyled>
`;

exports[`<Tooltip /> should render component with position bottom 1`] = `
<TooltipStyled
  content="Tooltip Text"
  position="bottom"
>
  <span>
    Hover me
  </span>
</TooltipStyled>
`;

exports[`<Tooltip /> should render component with position left 1`] = `
<TooltipStyled
  content="Tooltip Text"
  position="left"
>
  <span>
    Hover me
  </span>
</TooltipStyled>
`;

exports[`<Tooltip /> should render component with position right 1`] = `
<TooltipStyled
  content="Tooltip Text"
  position="right"
>
  <span>
    Hover me
  </span>
</TooltipStyled>
`;
